### DISCUSSION TITLE
HomeAssistant Component

### DISCUSSION THREAD

**Discussion by topic2k** [thomluther/anker-solix-api#52] [General]
Hello @thomluther,

i used your library (thank you for that) to build a custom component for HomeAssistant. I have a working version https://github.com/topic2k/anker-solix-api-homeassistant.
Now that i wanted to add your library as requirement, i had the pip install problem #51. While going through the issues, i read that you already have a HomeAssistantcomponent (nearly done). I'm eager to see your version.
I've used the component to learn a bit about creating custom components for HomeAssistant, so i don't care about having done unnessecary work.

**Comment by thomluther** [comment]
Hi @topic2k 
Yeah, I'm pretty much done with it and want to release that via hacs.
The api library is changing also with it to satisfy the needs for the integration, So that may break yours when json field names change.
I had a look to your code and that is good for a base start.
But I put lots more structure in it with automatic device details refresh build in, translations, icon customization etc.
Here is just a screenshot for a system device with some fake data
![image](https://github.com/thomluther/anker-solix-api/assets/97168515/314f8f27-7350-40a6-bed7-04543111419b)

It will be hopefully available by mid March.
Still needs more testing and value validations. I put some work arounds in the lib to avoid non-sense info as returned by the api.
We should also avoid too many integrations will start flowing around. The existing Add-on is already pretty difficult to use, it would just give bad user experience.

**Reply by topic2k** [reply]
Looks good. I'll stop working on my integration and remove it from public.

**Comment by thomluther** [comment]
Where I still need help that can be eventually explored with this api is:

- how to change home load, what options are available when submitting a new schedule or single time slot see #19 
- how to change power cut off
- how to see if firmware update is available? Can it be triggered via the api?
- how to get the temperature? I assume this might be an optional parameter with any of the device or site queries? I haven't found a specific endpoint for the temperature and it nowhere included 
-

**Reply by topic2k** [reply]
I don't own the system, i just have a family acces to a system of a friend, who wants to have it integrated into his Homeassistant setup. So it would be difficult to help, but when i have instruction for maybe to run a datalogger or something, i can ask him.

**Reply by erichirschgh** [reply]
About your 4th bullet point above -- how to get the temperature?  Did you make any progress there, I would be interested in monitoring the temperature as well via the api approach.  If you have any thoughts or endpoints you think it might be worth exploring I would be willing to poke around a bit and see if I find anything.  Thanks for all the great work meanwhile!

**Reply by thomluther** [reply]
No, I don't think the temp will be reported to the cloud.
It is visible only on the device details page, and those data are typically provided via BT or the MQTT server.
Only the System home page data is typically available via Cloud Api

**Comment by thomluther** [comment]
Hi @topic2k 
Thanks for the offer. Exploring the api typically requires access with an owner account, otherwise there are too many permission restrictions any many queries don't work. Doesn't make sense to explore parameters with an account that has no device permissions anyway.
